dev-guide/cluster-fleet-evaluation.md

```diff
@@ -0,0 +1,181 @@
+---
+title: cluster-fleet-evaluation
+authors:
+  - "@aravindhp"
+reviewers:
+  - "@openshift/openshift-patch-managers"
+approvers:
+  - "@openshift/openshift-patch-managers"
+creation-date: 2022-08-16
+last-updated: 2022-08-16
+---
+
+# Cluster Fleet Evaluation
+
+## Problem Statement
+
+OpenShift components sometimes need to introduce a change that could result in
+highly visible alerts, breakages or upgrade failures on user clusters. In
+scenarios like this it will be useful for the component owner to assess how many
+customers would be affected before rolling out the change and based on that
+feedback decide on the level of investment required to properly inform the
+customer base, or change implementation details.
+
+## Solution
+
+Rather than introducing the change, add just the detection logic to the operator
+for the change and associate it with a
+[ClusterStatusConditionTypes](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L145)
+called [EvaluationConditionDetected](https://github.com/openshift/api/pull/1250).
+If the
+[ConditionStatus](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L124)
+associated with this type is
+[ConditionTrue](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L107)
+and the includes the
+[Reason]([Reason](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L133))
+you are expecting, it will indicate that the change would have affected that
+cluster. However, if the _ConditionStatus_ is _false_, or it's _true_ but the
+_Reason_ does not include the _Reason_ you are expecting, the change doesn't
+affect the cluster.  You can join multiple _Reason_ to test multiple scenarios.
```

Comment:
What's the syntax for joining reasons?

Reply:
I think it is `::`. @deads2k please confirm.